### PR TITLE
Fix #204 - Update MongoDB to remove deprecation warnings

### LINKED ISSUE #204: DeprecationWarning: collection.count using mongo 4.X
```
iot-iotagent-manager        | (node:16) DeprecationWarning: collection.count is deprecated, and will be removed in a future version. Use Collection.countDocuments or Collection.estimatedDocumentCount instead
``

### PR DESCRIPTION
**Pull request by jason-fox**
- Update Drivers
- set `useUnifiedTopology`
- count => `countDocuments()`
-  remove => `deleteOne()`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "husky": "~4.2.5",
     "lint-staged": "~10.2.11",
     "mocha": "8.0.1",
-    "mongodb": "3.5.9",
+    "mongodb": "3.6.4",
```
Comment (fgalan) [inline package.json:48]: Should be noted in CHANGES_NEXT_RELEASE, this way:

```
- Upgrade mongodb dev dependency from 3.5.9 to 3.6.4 (#204)
```

(I'm not sure about citing #204 in this case...)

File: package.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
     "logops": "2.1.0",
     "underscore": "~1.7.0",
     "revalidator": "~0.3.1",
-    "mongoose": "5.7.7",
+    "mongoose": "5.11.18",
```
Comment (fgalan) [inline package.json:37]: Should be noted in CHANGES_NEXT_RELEASE, this way:

```
- Upgrade mongoose dependency from 5.7.7 to 5.11.18 (#204)
```

Comment (jason-fox) [reply package.json:37]: Fixed 31dfddf

File: package.json (hunk 1 above)
Comment (jason-fox) [reply package.json:48]: Fixed 31dfddf

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**